src/calcium/ca.jl

```diff
@@ -0,0 +1,1447 @@
+   check_special(r)
+   return r
+end
+
++(a::ca, b::qqbar) = a + parent(a)(b)
+
+function -(a::ca, b::ca)
+   a, b = same_parent(a, b)
+   C = a.parent
+   r = C()
+   ccall((:ca_sub, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function -(a::ca, b::Int)
+   C = a.parent
+   r = C()
+   ccall((:ca_sub_si, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Int, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function -(a::ca, b::fmpz)
+   C = a.parent
+   r = C()
+   ccall((:ca_sub_fmpz, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpz}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function -(a::ca, b::fmpq)
+   C = a.parent
+   r = C()
+   ccall((:ca_sub_fmpq, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpq}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+-(a::ca, b::qqbar) = a - parent(a)(b)
+
+function -(a::Int, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_si_sub, libcalcium), Nothing,
+         (Ref{ca}, Int, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function -(a::fmpz, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_fmpz_sub, libcalcium), Nothing,
+         (Ref{ca}, Ref{fmpz}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function -(a::fmpq, b::ca)
+   C = b.parent
+   r = C()
+   ccall((:ca_fmpq_sub, libcalcium), Nothing,
+         (Ref{ca}, Ref{fmpq}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+-(a::qqbar, b::ca) = parent(b)(a) - b
+
+function *(a::ca, b::ca)
+   a, b = same_parent(a, b)
+   C = a.parent
+   r = C()
+   ccall((:ca_mul, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function *(a::ca, b::Int)
+   C = a.parent
+   r = C()
+   ccall((:ca_mul_si, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Int, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function *(a::ca, b::fmpz)
+   C = a.parent
+   r = C()
+   ccall((:ca_mul_fmpz, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpz}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+function *(a::ca, b::fmpq)
+   C = a.parent
+   r = C()
+   ccall((:ca_mul_fmpq, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{fmpq}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
+*(a::ca, b::qqbar) = a * parent(a)(b)
+
+*(a::Int, b::ca) = b * a
+*(a::fmpz, b::ca) = b * a
+*(a::fmpq, b::ca) = b * a
+*(a::qqbar, b::ca) = b * a
+
+
+function //(a::ca, b::ca)
+   a, b = same_parent(a, b)
+   C = a.parent
+   r = C()
+   ccall((:ca_div, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, b, C)
+   check_special(r)
+   return r
+end
+
```

Comment:
Could you separate into sections Ad hoc division, Powering, Unary operations, for conformity with the other files.

Reply:
Ok.